src/cpu/x64/jit_uni_reorder.cpp

```diff
@@ -902,6 +904,341 @@ struct jit_uni_reorder_kernel_f32 : public kernel_t, public jit_generator {
     Zmm bf16_emu_reserv_4 = Zmm(19);
 };
 
+// Seperate class for no unroll/threading burden
+struct jit_single_blk_kernel : public jit_generator {
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(jit_single_blk_kernel)
+    static bool applicable(const prb_t &p) {
+        using namespace data_type;
+
+        bool ok = true && p.ndims >= 2 && mayiuse(avx2)
+                && p.scale_type == scale_type_t::NONE
+                && utils::one_of(p.itype, f32 /*, bf16, s32, s8, u8*/)
+                && utils::one_of(p.otype, f32 /*, bf16, s32, s8, u8*/)
+                && IMPLICATION(
+                        p.itype == bf16, utils::one_of(p.otype, f32, bf16))
+                && IMPLICATION(
+                        p.otype == bf16, utils::one_of(p.itype, f32, bf16))
```

Comment:
unclear, is bf16 supported or not. Please simplify the checks here. Please remove `/* ... */` if the values are not supported.